routines/SAMIHOM4.m

```diff
@@ -143,11 +143,217 @@
  ;
  i route="report" d  q  ; 
  . m SAMIARG=vars
- . d WSREPORT^SAMIUR(.SAMIRESULT,.SAMIARG)
+ . d WSREPORT^SAMIUR(.SAMIRESULT,.vars)
  ;
+ i route="addperson" d  q  ;
+ . m SAMIARG=vars
+ . n form
+ . s form="vapals:addperson"
+ . d RTNPAGE^SAMIHOM4(.SAMIRESULT,form,.SAMIARG) q  ;
+ ;
+ i route="editperson" d  q  ;
+ . m SAMIARG=vars
+ . n sid s sid=$g(vars("studyid")) ; must have a sid
+ . i sid="" d  q  ;
+ . . d GETHOME^SAMIHOM3(.SAMIRESULT,.SAMIARG) ; on error go home 
+ . n sien s sien=$$SID2NUM^SAMIHOM3(sid)
+ . i sien="" d  q  ;
+ . . d GETHOME^SAMIHOM3(.SAMIRESULT,.SAMIARG) ; on error go home 
+ . n root s root=$$setroot^%wd("vapals-patients")
+ . s vars("saminame")=$g(@root@(sien,"saminame"))
+ . s vars("dob")=$g(@root@(sien,"dob"))
+ . s vars("sbdob")=$g(@root@(sien,"dob"))
+ . s vars("gender")=$g(@root@(sien,"gender"))
+ . s vars("sex")=$g(@root@(sien,"sex"))
+ . s vars("icn")=$g(@root@(sien,"icn"))
+ . s vars("ssn")=$g(@root@(sien,"ssn"))
+ . s vars("last5")=$g(@root@(sien,"last5"))
+ . s vars("dfn")=$g(@root@(sien,"dfn"))
+ . m SAMIARG=vars
+ . n form,err,zhtml
+ . s form="vapals:addperson"
+ . d RTNPAGE^SAMIHOM4(.SAMIRESULT,form,.SAMIARG) q  ;
+ ;
+ i route="register" d  q  ;
+ . m SAMIARG=vars
+ . d REG^SAMIHOM4(.SAMIRESULT,.SAMIARG)
+ ; 
  quit  ; End of WSVAPALS
  ;
  ;
+REG(SAMIRTN,SAMIARG) ; manual registration
+ ;
+ m ^gpl("reg")=SAMIARG
```

Comment:
Is this just for debugging? Do you really want this to happen in production?

Reply:
Tagging @glilly to respond

Reply:
If a problem happens in production, it might give us a clue as to what happened. So let's leave it there for now. gpl